static/js/lezer-parsers/language-packages.ts

```diff
@@ -0,0 +1,241 @@
+import { parser as level1Parser} from './level1-parser'
+import { parser as level2Parser} from './level2-parser'
+import { parser as level3Parser} from './level3-parser'
+import { parser as level4Parser} from './level4-parser'
+import { parser as level5Parser} from './level5-parser'
+import { parser as level6Parser} from './level6-parser'
+import { parser as level7Parser } from './level7-parser'
+import { parser as level8Parser } from './level8-parser'
+import { parser as level10Parser } from './level10-parser'
+import { parser as level11Parser } from './level11-parser'
+import { parser as level12Parser } from './level12-parser'
+import { parser as level13Parser } from './level13-parser'
+import { parser as level14Parser } from './level14-parser'
+import { parser as level15Parser } from './level15-parser'
+import { parser as level16Parser } from './level16-parser'
+import { parser as level17Parser } from './level17-parser'
+import { parser as level18Parser} from './level18-parser'
+import { tags as t } from "@lezer/highlight";
+import { LRParser } from '@lezer/lr';
+
+interface languageSupport {
+    parser: LRParser,
+    styleTags: Record<string, any>
+}
+
+export let languagePerLevel: Record<number, languageSupport> ={
+    1: {
+        parser: level1Parser,
+        styleTags: {
+            "print ask echo forward turn color": t.keyword,
+            "Text": t.name,
+            Comment: t.lineComment,
+            "Command/ErrorInvalid/Text": t.invalid,
+        }
```

Comment:
There's a bunch of repetition in these `styleTags`. I assume that the values have to be the same in every level?

Is it required that we only define the keys that actually exist in every level, or is it possible to just define a full map of everything?

If it is really the case that we must only supply the keys that are actually used, then we could use some additional information to filter down from the big map?

Something like this (but not exactly this):

```ts
const ALL_STYLE_TAGS = {
          "print forward turn color ask is echo sleep Op at random remove from add to Comma": t.keyword,   
          Comment: t.lineComment,
          "Text": t.name,
          "Command/ErrorInvalid/Text": t.invalid,
      };


function specificStyleTags(...tagNames: string[]): Record<string, any> {
  // flatMap allows us to map and filter at the same time
  return Object.fromEntries(Object.entries(ALL_STYLE_TAGS).flatMap((namesString, token) => {
    const names = namesString.split(/\s+/).filter(n => tagNames.includes(n));
    if (names.length === 0) {
      // Nothing left
      return [];
    }
    // the 'as const' here is because I'm not sure TypeScript will typecheck this correctly without it
    return [[names, token] as const];
  });
}
```

Reply:
You were right that you can supply keys that aren't actually used. Took advantage of that and simplified this a lot!